lib/libzfs/libzfs_status.c

```diff
@@ -51,27 +51,41 @@
 
 /*
  * Message ID table.  This must be kept in sync with the ZPOOL_STATUS_* defines
- * in libzfs.h.  Note that there are some status results which go past the end
- * of this table, and hence have no associated message ID.
+ * in include/libzfs.h.  Note that there are some status results which go past
+ * the end of this table, and hence have no associated message ID.
  */
 static char *zfs_msgid_table[] = {
-	"ZFS-8000-14",
-	"ZFS-8000-2Q",
-	"ZFS-8000-3C",
-	"ZFS-8000-4J",
-	"ZFS-8000-5E",
-	"ZFS-8000-6X",
-	"ZFS-8000-72",
-	"ZFS-8000-8A",
-	"ZFS-8000-9P",
-	"ZFS-8000-A5",
-	"ZFS-8000-EY",
-	"ZFS-8000-EY",
-	"ZFS-8000-EY",
-	"ZFS-8000-HC",
-	"ZFS-8000-JQ",
-	"ZFS-8000-K4",
-	"ZFS-8000-ER",
+	"ZFS-8000-14", /* ZPOOL_STATUS_CORRUPT_CACHE */
+	"ZFS-8000-2Q", /* ZPOOL_STATUS_MISSING_DEV_R */
+	"ZFS-8000-3C", /* ZPOOL_STATUS_MISSING_DEV_NR */
+	"ZFS-8000-4J", /* ZPOOL_STATUS_CORRUPT_LABEL_R */
+	"ZFS-8000-5E", /* ZPOOL_STATUS_CORRUPT_LABEL_NR */
+	"ZFS-8000-6X", /* ZPOOL_STATUS_BAD_GUID_SUM */
+	"ZFS-8000-72", /* ZPOOL_STATUS_CORRUPT_POOL */
+	"ZFS-8000-8A", /* ZPOOL_STATUS_CORRUPT_DATA */
+	"ZFS-8000-9P", /* ZPOOL_STATUS_FAILING_DEV */
+	"ZFS-8000-A5", /* ZPOOL_STATUS_VERSION_NEWER */
+	"ZFS-8000-EY", /* ZPOOL_STATUS_HOSTID_MISMATCH */
+	"ZFS-8000-EY", /* ZPOOL_STATUS_HOSTID_ACTIVE */
+	"ZFS-8000-EY", /* ZPOOL_STATUS_HOSTID_REQUIRED */
+	"ZFS-8000-HC", /* ZPOOL_STATUS_IO_FAILURE_WAIT */
+	"ZFS-8000-JQ", /* ZPOOL_STATUS_IO_FAILURE_CONTINUE */
+	"ZFS-8000-MM", /* ZPOOL_STATUS_IO_FAILURE_MMP */
```

Comment:
question: is it ok to come up with new msgids "manually"? Or are they generated by some kind or algorithm/tool for which the FMA Message Registry (https://www.illumos.org/msg) is the frontend?

See for instance this comment:

https://github.com/illumos/illumos-gate/blob/e88f57b4cab0ed8f8a773d404d6781e20a4c26d7/usr/src/cmd/fm/dicts/ZFS.po#L24-L26

Reply:
If that's the case we can get a proper one and change it, I only picked MM as a stop-gap to fix the issue at hand since MMP has been in ZOL since 0.7

Reply:
They're clearly part of the the Solaris FMA infrastructure but I'm not sure exactly how the existing msgids were chosen.  `ZFS-8000-MM` doesn't appear to be used according to the FMA Message Registry link you posted, within ZoL itself there's no issue manually assigning these.

I know @ikozhukhov was recently looking in to porting MMP to dilos so perhaps he knows.  He'll also need to add this finalized patch to his port.

Reply:
do you have linux logic similar to illumos FMA for ZFS recovery?

Reply:
Parts of it yes, the core ZFS specific Solaris FMA logic was ported to ZoL and updated.  It's built in to a userspace daemon called the ZFS Event Daemon (ZED) and which behaves much like it's Solaris counterpart.  However, we didn't bring over all of FMA only the portions we needed for ZFS.

Reply:
thanks for update.
in this case, ID should be used/generated in the same model as solaris FMA for ZFS needs.

Reply:
I wouldn't know how to go about obtaining a new msgid, would it be possible for someone get one for us and let me know what to change it to on our end?  Thanks

Reply:
Sorry, i did not intend to derail this PR, i was only worried this msgid could be used in Illumos for something totally different in the future.

Reply:
to be safe in this way, we can contribute reservation of ID to illumos.
this way has been used in others shared IDs between platforms.

Reply:
OK, after some additional discussion we've decided to merge this change as is which is fine for Linux.  We can investigate the best way to sync up additional FMA changes independently.